src/components/record/related-table-actions.tsx

```diff
@@ -445,101 +531,331 @@ const RelatedTableActions = ({
+
+            if (totalWidth + buttonLeftOffset + buttonWidth <= containerWidth) {
+              // calculate the total button container width if it doesn't overflow
+              totalWidth += buttonWidth;
+              if(buttonContainer.getBoundingClientRect().height === 0){
+                tableHeaderButtonsToAddToDropdown.push(button);
+              }
+            } else {
+              // Push to dropdown if it overflowa
+              tableHeaderButtonsToAddToDropdown.push(button);
+            }
+          });
+          setButtonsInDropdown(tableHeaderButtonsToAddToDropdown);
         }
-        {allowCustomModeRelated(relatedModel) && renderCustomModeBtn()}
-        <ChaiseTooltip
-          placement='top'
-          tooltip={<span>Explore more {currentTable} records related to this {mainTable}.</span>}
-        >
-          <a className='chaise-btn chaise-btn-secondary more-results-link' href={exploreLink} onClick={onExplore}>
-            <span className='chaise-btn-icon fa-solid fa-magnifying-glass'></span>
-            <span>Explore</span>
-          </a>
-        </ChaiseTooltip>
+    };
+    calculateButtons();
+
+    const mainResizeSensor = new ResizeSensor(containerRef.current as Element, () => {
+      calculateButtons();
+    });
+
+    return () => {
+      mainResizeSensor.detach();
+    };
+  }, [isMobile]);
+
+  // This function is to toggle the dropdown to show or hide with options
+  const toggleDropdown = (bool: boolean, evt: any) => {
+    evt.originalEvent.stopPropagation();
+    setIsDropdownOpen(!isDropdownOpen);
+  };
+
+  // Adding different classname for table header and main section to apply styles
+  const containerClassName = `related-table-actions ${isMobile ? 'is-mobile ' : ''}${
+    !relatedModel.isInline ? 'table-header-actions' : 'main-section-actions'
+  }`;
+
+  /*
+   * Callback function for dropdown item click
+   */
+  const handleDropDownClick = (evt: any, button: any) => {
+    const buttonText = button.textContent;
+    evt.stopPropagation();
+    switch (buttonText) {
+      case 'Explore':
+        onExplore(evt);
+        break;
+      case 'Link records':
+      case 'Add records':
+        onCreate(evt);
+        break;
+      case 'Edit Records':
+        onUnlink(evt);
+        break;
+      case 'Unlink records':
+        onUnlink(evt);
+        break;
+      case 'Edit mode':
+      case 'Custom mode':
+      case 'Table mode':
+        onToggleDisplayMode(evt);
+        break;
+    }
+  };
+  return (
+    <>
+      <div className={containerClassName} ref={containerRef}>
+        {renderButtons()}
+        {buttonsInDropdown.length > 0 && (
+          <Dropdown onToggle={(isOpen: boolean, event: any) => toggleDropdown(isOpen, event)}>
+            <Dropdown.Toggle
+              disabled={false}
+              className={`chaise-btn chaise-btn-primary dropdown-toggle-table ${isMobile
+                && !relatedModel.isInline ? 'dropdown-toggle-table-mobile' : ''}`}
```

Comment:
Change `dropdown-toggle-table-mobile` to something else to be consistent with the new state variable name.